### ISSUE TITLE
Script action format and task

### ISSUE THREAD

**Issue by BoPeng** [BoPeng/SOS#133]
Patch https://github.com/BoPeng/SOS/commit/a27551f587acea1e8e90db0c1668993d7e37058e in the [task branch](https://github.com/BoPeng/SOS/tree/task) implements the alternative script format action behavior.

Pros:

* clear explanation. 

```
 action: option
    script
```

is just an alias to

```
action(script, option)
```

* No mixing up of runtime and action options with

```
task: concurrent=True, workdir='/path/to/dir'
Rmarkdown: output_dir='/output'
   script
```

Cons:

* Script is a bit longer,

```
run: concurrent=True
   script
```
looks better than

```
task: concurrent=True
run:
   script
```

* There is still a need for inline task with `concurrent` or `workdir`, so  `task: workdir='/some/path` implying external task is not that convincing or clear.

It perhaps makes sense to define `concurrent`, `workdir`, `docker_image` etc as **common action options**, and leave `task` with external only options (there is none right now). At least in this way we can still enjoy

```
run: concurrent=True
   script
```

However, conceptually speaking, the run action should not control other actions in terms of `concurrent`. Even if we separate runtime and action options, it is likely that we have

* `workdir`, `docker_image`, `docker_file` as action options
* `workdir`, `concurrent` as runtime (task) options

With all factors considered, the existing behavior might not be too bad.

**Comment by BoPeng** [comment]
There is another argument here related to script format. Right now we cannot do

```
%%sos
run: 
   echo "something"
```

in iPython because sos takes this as part of a global section (no section name), which disallows directive (in this case task). If we separate script format with the task business, we would be able to use script format in global section, and do things like

```
report:
   hey, this is the header of my report
```

Note that we do not encourage calling functions in global section, but writing report does not cost much, even if it will be executed a few times.

The bottom line is that mixing script format (a format issue) with task (a runtime issue) seems to be silly.

**Comment by gaow** [comment]
I believe this is also related to my most recent comment to #132 where `concurrent` can be reset if we allow for the separation.

**Comment by BoPeng** [comment]
```
%%sos
run: 
   echo "something"
```

works, as well as

```
sospaste
```

This is very important because we can then execute pieces of shell or R code in ipython.

Perhaps we can attract some users just for this ability. :-)

**Comment by gaow** [comment]
I assume yes, particularly for bioinformaticists who does not regularly use interactive environment like RStudio. I believe this feature should go to the "modern bioinformatics" section of the paper :)
